src/app.c

```diff
@@ -213,18 +218,28 @@ bool MTY_WindowSetUITexture(MTY_App *app, MTY_Window window, uint32_t id, const
 		return false;
 
 	MTY_Device *device = gfx_ctx_get_device(cmn);
+	if (!device)
+		return false;
 
 	bool r = gfx_begin_ui(cmn, device);
 
 	if (r) {
-		void *texture = MTY_HashPopInt(cmn->ui_textures, id);
-		GFX_UI_API[cmn->api].destroy_texture(cmn->gfx_ui, &texture, device);
+		void *texture = MTY_HashGetInt(cmn->ui_textures, id);
+
+		if (rgba) {
+			void *texture_new = GFX_UI_API[cmn->api].create_texture(cmn->gfx_ui, device, rgba, width, height);
 
-		if (rgba)
-			texture = GFX_UI_API[cmn->api].create_texture(cmn->gfx_ui, device, rgba, width, height);
+			if (texture_new) {
+				GFX_UI_API[cmn->api].destroy_texture(cmn->gfx_ui, &texture, device);
```

Comment:
Please check if `texture` is NULL before destroying it. An up-front check is ideal, even if the individual APIs may check too